app/models/grda_warehouse/auth_policies/project_pii_policy.rb

```diff
@@ -0,0 +1,37 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# FIXME: in a report context we may only have a project id to determine PII. Using just the project id is efficient but
+# the # permissions not are entirely accurate in every case. We are not including permissions granted by all the
+# client's enrollments or direct client-record visibility
```

Comment:
Would the fix for this be to convert usage to a ProjectPolicy or ClientPolicy instead?  Basically batch fetch the actual resources, index on `id` and then act on the appropriate resource instead of the id?